src/pixel/with_alpha.rs

```diff
@@ -0,0 +1,87 @@
+use crate::*;
+
+/// A pixel which can gain an alpha component.
```

Comment:
maybe instead of gain/lose, this could be has/hasn't, and offer `fn alpha(self) -> T` without `Option`?

Reply:
The gain/lose could still be a useful conversion for some usecases. Perhaps we rename this trait to `GainAlpha` and create another trait `WithAlpha` which has `fn alpha(&self) -> Self::AlphaComponent`?

Reply:
I've renamed them to `{Gain,Lose}Alpha` and added an `HasAlpha` trait and renamed the `alpha()` function on `HeterogeneousPixel` to `alpha_checked()` to prevent name ambiguity.